### ISSUE TITLE
Tokens not being cached properly - 'No matching token found when filtering by scope'

### ISSUE THREAD

**Issue by jacksonhart** [AzureAD/microsoft-authentication-library-for-js#2231]
Bug Report / Configuration Help

Chrome 84.0.4147.135

msal-angular 1.1.1
msal 1.3.3
angular 9.1.12

I am getting 'No matching token found when filtering by scope' causing a token request on each page route instead of fetching it from the cache.
I am not intentinally trying to acquire tokens for scopes that span multiple resources as mentioned in [1240](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/1240).

UserAgentApplication.getCachedToken() seems to be called once with a request that has my app/API scope. In this case the cached token is found.
But then it is called again with a request with scopes ["openid","profile"] which is not found in the cache.

In localstorage I have 2 token entries, one with my app/API scope and one **with no scopes**.
I am assuming this is relevant - the key has clientId, homeAccountIdentifier, authority but no 'scope' attribute.

`environment.consentScopes: ['https://XXX.onmicrosoft.com/XXXWebApiDev/Access']`

````
MsalModule.forRoot({
    auth: {
        clientId: environment.clientID,
        authority: environment.authority,
        redirectUri: environment.redirectUri,
        postLogoutRedirectUri: environment.redirectUri,
        navigateToLoginRequestUrl: false,
        validateAuthority: false,
      },
      cache: {
        cacheLocation: 'localStorage',
        storeAuthStateInCookie: true
      },
      system: {
        logger: new Logger(loggerCallback, {
            correlationId: '1234',
            level: LogLevel.Verbose,
            piiLoggingEnabled: true,
        }),
    }
    },
    {
      popUp: false,
      consentScopes: ['openid', 'profile'], // also tried environment.consentScopes and leaving blank
      protectedResourceMap = [
          ['https://graph.microsoft.com/v1.0/me', ['user.read']],
          [environment.apiUrl, environment.consentScopes]
      ],
      extraQueryParameters: {}
    })
```

**Comment by jmckennon** [comment]
@jacksonhart Are you using the `msal-guard` on your routes?

**Comment by jacksonhart** [comment]
Hi @jmckennon yes we are using it

**Comment by jmckennon** [comment]
This problem is likely addressed by #2206. That PR should be merged in the near future. If you're still seeing issues after it's been released please let us know.

**Comment by jacksonhart** [comment]
Thanks. downgraded msal and its working .
Will close this issue.
